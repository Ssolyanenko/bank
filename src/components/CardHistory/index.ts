export * from './History';
export * from './Pagination';
export * from './CardHistory';
