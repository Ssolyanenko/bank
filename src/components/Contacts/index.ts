export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
