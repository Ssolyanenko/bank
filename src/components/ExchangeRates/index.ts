export * from './ExchangeRates';
export * from './Converter';
export * from './Rates';
