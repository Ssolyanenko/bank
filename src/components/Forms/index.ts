export * from './Login';
export * from './PasswordRecoveryForm';
export * from './GeneralInfo';
export * from './Security';
export * from './OrderDebitTravelCard';
export * from './SetPinCodeForm';
export * from './OrderCreditCardPremium';
export * from './FormNotifications';
export * from './FilterHistoryForm';
export * from './CardLimitForm';
export * from './OrderCreditCardBilliable';
export * from './OrderDebitSmartCard';
export * from './TransferByPhoneNumberForm';
export * from './AcceptTerms';
