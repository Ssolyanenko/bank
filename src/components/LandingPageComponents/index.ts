export * from './LandingPageHeader';
export * from './LandingPageMain';
export * from './LandingPageFooter';
