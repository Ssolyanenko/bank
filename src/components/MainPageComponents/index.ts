export * from './FeaturedServices';
export * from './Menu';
export * from './DebitCards';
export * from './OperationHistory';
export * from './CreditCards';
export * from './CardProductsInner';
export * from './DebitSmartCard';
export * from './DebitTravelCard';
export * from './TransferList';
