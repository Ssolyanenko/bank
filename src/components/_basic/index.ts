export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
