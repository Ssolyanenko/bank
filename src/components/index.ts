export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
