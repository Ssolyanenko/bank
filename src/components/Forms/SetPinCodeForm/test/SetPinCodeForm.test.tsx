import 'jsdom-global/register';
import React from 'react';
import { mount, shallow } from 'enzyme';

import { SetPinCodeForm } from 'components';
import { ButtonNames } from 'constants/text';

describe('<SetPinCodeForm />', (): void => {
  test('should make snapshot of the component', (): void => {
    const props = {
      handlePinChangeModal: jest.fn(),
      title: 'Some title',
      buttonName: ButtonNames.SET_PIN_CODE,
    };

    const wrapper = shallow(<SetPinCodeForm {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should change field label on click', (): void => {
    let isFieldClickedMock = false;
    const setIsFieldClickedMock = jest.fn((x) => (isFieldClickedMock = x));
    jest.spyOn(React, 'useState').mockImplementation(() => [isFieldClickedMock, setIsFieldClickedMock]);

    const props = {
      handlePinChangeModal: jest.fn(),
      title: 'Some title',
      buttonName: ButtonNames.SET_PIN_CODE,
    };

    const wrapper = mount(<SetPinCodeForm {...props} />);

    wrapper.find('label.inputLabel').at(0).simulate('click');
    expect(setIsFieldClickedMock).toHaveBeenCalled();
  });

  test('should change newPin value', (): void => {
    const handleValueChange = jest.fn();

    const props = {
      handlePinChangeModal: jest.fn(),
      title: 'Some title',
      buttonName: ButtonNames.SET_PIN_CODE,
      handleValueChange: handleValueChange,
    };

    const wrapper = mount(<SetPinCodeForm {...props} />);

    wrapper.find('input[name="newPin"]').simulate('change', { target: { value: '1111' } });
    expect(handleValueChange).toHaveBeenCalled();
  });

  test('should change newPinConfirm value', (): void => {
    const handleValueChange = jest.fn();

    const props = {
      handlePinChangeModal: jest.fn(),
      title: 'Some title',
      buttonName: ButtonNames.SET_PIN_CODE,
      handleValueChange: handleValueChange,
    };

    const wrapper = mount(<SetPinCodeForm {...props} />);

    wrapper.find('input[name="newPinConfirm"]').simulate('change', { target: { value: '1111' } });
    expect(handleValueChange).toHaveBeenCalled();
  });
});
