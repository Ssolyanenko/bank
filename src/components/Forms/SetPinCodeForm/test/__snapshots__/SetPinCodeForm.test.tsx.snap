// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SetPinCodeForm /> should make snapshot of the component 1`] = `
<Formik
  initialValues={
    {
      "newPin": "",
      "newPinConfirm": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_excludedEdges": [],
      "_mutate": undefined,
      "_nodes": [
        "newPin",
        "newPinConfirm",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "conditions": [],
      "deps": [],
      "exclusiveTests": {},
      "fields": {
        "newPin": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": [],
          "deps": [],
          "exclusiveTests": {
            "matches": false,
            "min": true,
            "required": true,
          },
          "spec": {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": [
            [Function],
          ],
          "type": "string",
        },
        "newPinConfirm": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": [
            Condition {
              "fn": [Function],
              "refs": [
                Reference {
                  "getter": [Function],
                  "isContext": false,
                  "isSibling": true,
                  "isValue": false,
                  "key": "newPin",
                  "map": undefined,
                  "path": "newPin",
                },
              ],
            },
          ],
          "deps": [
            "newPin",
          ],
          "exclusiveTests": {
            "matches": false,
            "min": true,
            "required": true,
          },
          "spec": {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": [
            [Function],
          ],
          "type": "string",
        },
      },
      "spec": {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": [],
      "transforms": [
        [Function],
      ],
      "type": "object",
    }
  }
>
  [Function]
</Formik>
`;
